### ISSUE TITLE
Support dated vs undated references

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-ieee#29]
Since the relaton-ieee fetches documents from the relaton-data-ieee we have to use references that match files in the source repo (IEEE 2830-2021 not IEEE 2830). 

1. We need to support undated vs dated references, by creating the undated reference as a collection of the published editions.
2. We need to implement indexing to be able using partly matching references

We should use `pubid-ieee` to handle this.

_Originally posted by @ronaldtse in https://github.com/relaton/relaton-ieee/issues/27#issuecomment-1196221667_

**Comment by andrew2net** [comment]
> We should use `pubid-ieee` to handle this.

@ronaldtse It seems the pubid-ieee handles only IEEE publisher but in the ieee-rawbib dataset there are many others (AIEE, ANSI, IEC, IEC/IEEE, IRE, ISO/IEC, ISO/IEC/IEEE, ISO/IEEE, NACE, NACE/IEEE, USAS). Do we need to support all these publishers?

**Comment by ronaldtse** [comment]
pubid-ieee supports most of those publishers from the ieee-rawbib dataset, and the remaining ones will be supported.

Yes we need to support all those publishers, but for example, the authoritative sources of IEC and ISO/IEC should not be the relaton-ieee processor.

This means that Relaton, when receiving "IEC", should be routed to relaton-iec. If the user specifies the relaton-ieee processor, then it uses relaton-ieee.

**Comment by andrew2net** [comment]
@ronaldtse you are right, I've tried it and it supports most of the publishers. But the gem isn't released yet. ping @mico

**Comment by ronaldtse** [comment]
> But the gem isn't released yet.

Oh my. Yes we need to do that!
